wellcomecollection/platform-infrastructure
# Upgrade cache invalidation lambdas to node20

## What does this change?

https://github.com/wellcomecollection/platform/issues/5837
Upgrades cloudfront cache invalidation lambdas to node20
Changes to the handler as the sdk used previously is not available in node20 lambdas

## How to test

In the platform account, publish invalidation request from the `api-stage-cloudfront-invalidate` topic
`{ "paths": "/catalogue/v2/works/z869w74a" }`
In the cloudfront console, go to distro `E9GLJ5CQAQPUK` (Collection APIs (stage)) > Invalidations tab. You should see the invalidation successfully completed here
In cloudwatch, log group `/aws/lambda/api-stage-cloudfront_invalidation` there should be no error.

NOTE: only the terraform **_in the 1st commit_** is applied and a .zip of the updated lambda has been uploaded to `s3://wellcomecollection-platform-infra/lambdas/cloudfront_invalidation/handler_with_node20.zip`. This is only deployed to `api-stage-cloudfront_invalidation` 
NEXT STEP: once this is merged we deploy the updated handler code to the other invalidation lambdas (unsure if this happens automatically) and apply the TF in the 3rd commit to upgrade the runtime of the other invalidation lambdas. 

## How can we measure success?

New and shiny node runtime 

## Have we considered potential risks?

These lambdas are only triggered manually and sporadically.
